language/move-core/types/src/vm_status.rs

```diff
@@ -57,7 +57,7 @@ pub enum VMStatus {
     /// Indicates an error from the VM, e.g. OUT_OF_GAS, INVALID_AUTH_KEY, RET_TYPE_MISMATCH_ERROR
     /// etc.
     /// The code will neither EXECUTED nor ABORTED
-    Error(StatusCode),
+    Error(StatusCode, Option<String>),
```

Comment:
Do we need to roll out a backwards compatible version of this? Or change the serialization behaviors so we don't embed the String?

Reply:
Yes, but this will happen in aptos-core, not in this repo.

Reply:
How would we gate this on the aptos side? Aren't many of these errors thrown inside the verifier or interpreter?

Reply:
Yep, we don't. We are converting this to our representation. We can decide on our side what we do with this new piece of information.